File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def dfs(self , root: Optional[TreeNode] , curr) -> None:
        if root is None :
            return
        
        curr = curr * 10 + root.val

        if root.left is None and root.right is None :
            self.ans += curr
            return

        self.dfs(root.left , curr) 
        self.dfs(root.right , curr)
    
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.ans = 0

        self.dfs(root , 0)

        return self.ans

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans = 0;

    private void dfs(TreeNode root , int curr){
        if(root == null)
            return;

        curr = curr* 10 + root.val;

        if(root.left == null && root.right == null)
            ans += curr;

        dfs(root.left , curr);
        dfs(root.right , curr);
    }

    public int sumNumbers(TreeNode root) {
        if(root == null)
            return 0;

        dfs(root , 0);
        return ans;
    }
}